From 4746722090513b5ad66819a758004ac198ccc56f Mon Sep 17 00:00:00 2001
From: Ken Zhu <quic_guigenz@quicinc.com>
Date: Thu, 28 Sep 2023 10:44:56 -0700
Subject: [PATCH] linux: fix dscpremark extention doesn't set the type length

It needs to set up the type length for each extension.

Change-Id: Ia2087d7c4653045665c006ca9d637ff49d27fd92
Signed-off-by: Ken Zhu <quic_guigenz@quicinc.com>
---
 net/netfilter/nf_conntrack_extend.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/net/netfilter/nf_conntrack_extend.c b/net/netfilter/nf_conntrack_extend.c
index 0b513f7bf9f3..ee3dbb2fe50b 100644
--- a/net/netfilter/nf_conntrack_extend.c
+++ b/net/netfilter/nf_conntrack_extend.c
@@ -24,6 +24,7 @@
 #include <net/netfilter/nf_conntrack_synproxy.h>
 #include <net/netfilter/nf_conntrack_act_ct.h>
 #include <net/netfilter/nf_nat.h>
+#include <net/netfilter/nf_conntrack_dscpremark_ext.h>
 
 #define NF_CT_EXT_PREALLOC	128u /* conntrack events are on by default */
 
@@ -54,6 +55,10 @@ static const u8 nf_ct_ext_type_len[NF_CT_EXT_NUM] = {
 #if IS_ENABLED(CONFIG_NET_ACT_CT)
 	[NF_CT_EXT_ACT_CT] = sizeof(struct nf_conn_act_ct_ext),
 #endif
+#ifdef CONFIG_NF_CONNTRACK_DSCPREMARK_EXT
+	[NF_CT_EXT_DSCPREMARK] = sizeof(struct nf_ct_dscpremark_ext),
+#endif
+
 };
 
 static __always_inline unsigned int total_extension_size(void)
@@ -85,6 +90,9 @@ static __always_inline unsigned int total_extension_size(void)
 #endif
 #if IS_ENABLED(CONFIG_NET_ACT_CT)
 		+ sizeof(struct nf_conn_act_ct_ext)
+#endif
+#ifdef CONFIG_NF_CONNTRACK_DSCPREMARK_EXT
+		+ sizeof(struct nf_ct_dscpremark_ext)
 #endif
 	;
 }
-- 
2.34.1

